src/main/kotlin/no/nav/familie/ks/sak/kjerne/brev/begrunnelser/BegrunnelserForPeriodeContext.kt

```diff
@@ -351,37 +331,52 @@ class BegrunnelserForPeriodeContext(
-                        .filter { vilkårResultat ->
-                            val vilkårResultatIkkeOppfyltForPeriode = vilkårResultat.verdi.any { !it.erOppfylt() }
-                            val vilkårResultatOppfyltRettEtterPeriode =
-                                vilkårResultat.verdi.any { it.erOppfylt() && it.periodeTom?.toYearMonth() == vedtaksperiode.fom.toYearMonth() }
+                        .singleOrNull()
+                        ?.verdi
+                        ?.takeIf { vilkårResultater ->
+                            val erAlleVilkårOppfylt = vilkårResultater.all { it.erOppfylt() }
+                            val erVilkårResultatSomOpphørerRettFørPeriode =
+                                vilkårResultater.any { it.erOppfylt() && it.periodeTom?.toYearMonth() == vedtaksperiode.fom.toYearMonth() }
+
+                            !erAlleVilkårOppfylt || erVilkårResultatSomOpphørerRettFørPeriode
+                        } ?: emptyList()
+
+                val forskjøvedeVilkårResultaterSomOpphørerFraForrigePeriode = vilkårResultatTidslinjeForPerson.finnVilkårResultaterSomOpphørerRettFørPeriode()
 
-                            vilkårResultatIkkeOppfyltForPeriode || vilkårResultatOppfyltRettEtterPeriode
-                        }.flatMap { it.verdi }
+                val forskjøvedeVilkårResultaterSomIkkeErOppfylt = forskjøvedeVilkårResultaterSomIkkeErOppfyltEllerMistetIPeriode + forskjøvedeVilkårResultaterSomOpphørerFraForrigePeriode
 
-                if (forskøvedeVilkårResultaterSomIkkeErOppfyltEllerOppfyltRettEtterPeriode.isNotEmpty()) {
-                    Pair(person!!, forskøvedeVilkårResultaterSomIkkeErOppfyltEllerOppfyltRettEtterPeriode)
+                if (forskjøvedeVilkårResultaterSomIkkeErOppfylt.isNotEmpty()) {
+                    Pair(person, forskjøvedeVilkårResultaterSomIkkeErOppfylt)
                 } else {
                     null
                 }
             }.toMap().filterValues { it.isNotEmpty() }
 
-    private fun finnPersonerMedVilkårResultaterSomGjelderIPeriode(): Map<Person, List<VilkårResultat>> =
-        personResultater.tilFørskjøvetOppfylteVilkårResultatTidslinjeMap(personopplysningGrunnlag)
-            .mapNotNull { (aktør, tidslinje) ->
-                val person =
-                    personopplysningGrunnlag.personer.find { it.aktør.aktivFødselsnummer() == aktør.aktivFødselsnummer() }
-                        ?: return@mapNotNull null
+    private fun Tidslinje<List<VilkårResultat>>.finnVilkårResultaterSomOpphørerRettFørPeriode(): List<VilkårResultat> {
+        val forrigePeriode =
+            tilPerioderIkkeNull()
+                .singleOrNull { it.tom?.plusDays(1) == vedtaksperiode.fom }
+
+        return forrigePeriode?.verdi?.filter { vilkårResultat ->
+            vilkårResultat.periodeTom != null &&
+                forrigePeriode.fom != null &&
+                vilkårResultat.periodeTom!!.toYearMonth() == vedtaksperiode.fom.toYearMonth()
```

Comment:
Sikker på at vi kan ha noen nullverdier her? 
Eventuelt kunne vi hatt en `fun LocalDate?.tilLocalDateEllerTidenesMorgen = this ?: TIDENES_MORGEN`